### ISSUE TITLE
[Bug] Mixing frontmatter tags and content tags excludes one of the categories

### ISSUE THREAD

**Issue by ngraham20** [mjsumpter/obsidian-tag-page#16]
I have some notes with the tag in the metadata and others with the same tag in the content. When I do this, it appears that whichever group is smaller will not get generated. If there are more content tags, the frontmatter ones aren't generated, and if there are more frontmatter ones, then the content one isn't generated. The header for the Content tags is still there, but it's not populated.

Here's some example pages that I have:
```
# daily-note_Tags
---
tag-page-query: "#daily-note"
---
## Tag Content for #daily-note
## Files with #daily-note in frontmatter
- [[Daily]]
- [[2024-03-15]]
- [[2024-03-14]]
- [[2024-02-29]]
- [[2024-02-25]]
```

```
# Testpage
#daily-note
```

```
# 2024-03-15
---
tags:
- daily-note
---
```

**Comment by mjsumpter** [comment]
Hi @ngraham20 , thanks for submitting this!

I'll try to get on this asap - a few questions for clarification:

- Your expectation above is that `Testpage` should show up under `## Files with #daily-note in frontmatter`, correct?
- Additionally, from your example, I can't tell what you would expect to be under `## Tag Content for #daily-note`. Can you perhaps show me your expected output? (Assuming what your provided is what you are actually seeing)

**Comment by mjsumpter** [comment]
I'm either misunderstanding, or this is a specific edge case (which is worth addressing), but I just want to be sure we are on the same page about the problem. Here are three pages I created:

"TestPage1":
```
#daily-note with some other content.

Here is a new paragraph.
```

"TestPage2":
```
---
tags:
- daily-note
---
```

and "TestPage3 both":
```
---
tags:
- daily-note
---
#daily-note but both with front-matter and other content

Here is a new paragraph for both
```

This is what the resulting tag page looks like:
```
---
tag-page-query: "#daily-note"
---
## Tag Content for #daily-note
- **daily-note** but both with front-matter and other content [[TestPage3 both|*]]
- **daily-note** with some other content. [[TestPage1|*]]
## Files with #daily-note in frontmatter
- [[TestPage3 both]]
- [[TestPage2]]
```

- Pages 1 & 3 both end up in the first section because they have content associated with the tag, but 2 does not
- Pages 2 & 3 both end up in the second section because they have frontmatter tags, but 1 does not.

From my understanding of your problem, your test page:
```
# Testpage
#daily-note
```

Doesn't have any content closely associated with the tag itself. The plugin currently is "opinionated", in the sense it has the following behavior:
- It assumes that if your goal is to tag an entire page at the top level, that you use frontmatter tags, and will be bucketed that way
- Otherwise, tag content within a note is meant to be captured as a minimal unit of a thought or sentence, as to not create bloat in the aggregated file. What this means is the plugin captures a tag (#daily-note) that is within some sentence fragment (it tries to bookend the tag between punctuation and newlines). It also grabs any indented lines in a bullet structure that come after that fragment

The important point (I think) is the second one. In your example test page `#daily-note` is a unit all by itself, separated by newlines. As such, the content isn't captured by the plugin intentionally.

Am I understanding the problem correctly? If I am, and you still would like this to change, can you describe to me your use-case for wanting to have a tag work standalone on a line like this?

**Comment by ngraham20** [comment]
The specific issue I'm running into is that I have primarily written my tags at the top of my page but not in the frontmatter. These tags also do not have any content after them (essentially doing the same thing as the frontmatter but not in the metadata section itself). What I'm finding is that I have a specific tag that is aggregating all of my pages except a few.

The ones that are being correctly collected are all in the format

```
# Page title
#tag/subtag
#anothertag
#a-third-tag
```

but a specific subset of my pages I have the tags in the format
```
---
tags:
- tag/subtag
- anothertag
- a-third-tag
---
```

and for some reason, these specific pages are being left out of the aggregation.

**Comment by ngraham20** [comment]
I did find however, that the reverse case was a slightly different bug than I thought. If I have an empty note with only the tag, i.e.

notice the lack of newline after the tag
```
# Page title
#tag```

then that particular page is also not added to the list.

**Comment by mjsumpter** [comment]
Okay I gotchya 🤙 . Will get to work on this

**Comment by mjsumpter** [comment]
Hey @ngraham20 , I just released an update (v1.0.1) that I think may have unintentionally fixed this. Can you let me know if you are still seeing issues?

**Comment by ngraham20** [comment]
hey yeah, that worked! Thanks a bunch @mjsumpter!
